app_spec/test/files/offline-validation.js

```diff
@@ -0,0 +1,103 @@
+const { one } = require('../config')
+
+const delay = ms => new Promise(resolve => setTimeout(resolve, ms)) 
+
+module.exports = scenario => {
+
+      // scenario('Can retriev header entries using get_entry', async (s, t) => {
+      //   const { alice, bob } = await s.players({ alice: one, bob: one })
+      //   await alice.spawn()
+      //   await bob.spawn()
+
+      //   await s.consistency()
+
+      //   // alice publishes a memo. This is private but should still publish a header
+      //   const create_result = await alice.call('app', "blog", "create_memo", { content: "private memo" })
+      //   t.comment(JSON.stringify(create_result))
+      //   t.equal(create_result.Ok.length, 46)
+      //   await s.consistency()
+
+      //   // get all the chain header hashes and check if they are retrievable
+      //   const maybe_chain_header_hashes = await alice.call('app', "blog", "get_chain_header_hashes", {})
+      //   t.ok(maybe_chain_header_hashes.Ok)
+      //   let chain_header_hashes = maybe_chain_header_hashes.Ok
+      //   t.equal(chain_header_hashes.length, 4) // dna, agentId, cap grant, memo
+
+      //   t.comment(JSON.stringify(chain_header_hashes))
+      //   let chain_headers = []
+
+      //   await s.consistency()
+
+      //   for (let i=0; i< chain_header_hashes.length; i++) {
+      //       // can use get_post because it just returns a raw entry given a hash
+      //       let header_hash = chain_header_hashes[i]
+      //       t.comment(header_hash)
+
+      //       // check alice can retrieve their own header entries
+      //       let header_alice = await alice.call('app', "blog", "get_post", { post_address: header_hash })
+      //       t.ok(header_alice.Ok)
+
+      //       // check bob can retrieve alices header entries
+      //       let header_bob = await bob.call('app', "blog", "get_post", { post_address: header_hash })
+      //       t.ok(header_bob.Ok)
+
+      //       t.deepEqual(header_alice.Ok, header_bob.Ok)
+
+      //       chain_headers.push(header_bob.Ok)
+      //   }
+      //   t.comment(JSON.stringify(chain_headers))
+      // })
+
+      scenario('Can perform validation of an entry while the author is offline', async (s, t) => {
```

Comment:
could just use `scenario.only` to make this the only test, without commenting a bunch of stuff out